src/services/permissions-checker.js

```diff
@@ -3,20 +3,55 @@ const moment = require('moment');
 const VError = require('verror');
 const _ = require('lodash');
 const forestServerRequester = require('./forest-server-requester');
-const { perform } = require('./base-filters-parser');
+const { perform: parseFilters } = require('./base-filters-parser');
+const logger = require('./logger');
 
-let permissionsPerRendering = {};
+const EXPIRATION_IN_SECONDS = process.env.FOREST_PERMISSIONS_EXPIRATION_IN_SECONDS || 3600;
 
-function PermissionsChecker(
-  environmentSecret,
-  renderingId,
-  collectionName,
-  permissionName,
-  permissionInfos = undefined,
-) {
-  const EXPIRATION_IN_SECONDS = process.env.FOREST_PERMISSIONS_EXPIRATION_IN_SECONDS || 3600;
+class PermissionsChecker {
+  constructor(environmentSecret, renderingId) {
+    this.environmentSecret = environmentSecret;
+    this.renderingId = renderingId;
+  }
+
+  static expirationInSeconds = EXPIRATION_IN_SECONDS;
+
+  // This permissionsPerRendering is the cache, shared by all instances of PermissionsChecker.
+  static permissionsPerRendering = {};
+
+  static cleanCache() {
+    PermissionsChecker.permissionsPerRendering = {};
+  }
+
+  static getPermissionsData(renderingId) {
+    if (!PermissionsChecker.permissionsPerRendering[renderingId]) {
+      return null;
+    }
 
-  function isSmartActionAllowed(smartActionsPermissions) {
+    return PermissionsChecker.permissionsPerRendering[renderingId].data;
+  }
```

Comment:
Why don't you do a getPermissions function ? That allow you to get rid of `PermissionsChecker.permissionsPerRendering[renderingId]` that you have in some places